### PR TITLE
feat(html-closing-bracket-new-line): add rule

### LINKED ISSUE #590: feat: Add `svelte/brackets-same-line` rule
### Motivation

I don't like brackets on their own line as it makes the code much longer and harder to read. It may be useful for editing but it's just messy overall. Sometimes use of VSCode's built-in formatter or projects that use to use prettier can have a lot of brackets on new lines. This rule makes it easy to clean things up.

### Description

Keep closing brackets on the same line as the last attribute or the tag itself

### Examples

```svelte
<script>
</script>

<!-- ✓ GOOD -->
<div>Text</div>

<div title="foo">Text</div>

<div 
  bar 
  title="foo">Text</div>

<div>
  Text
</div>

<div />

<!-- ✗ BAD -->
<div
  bar
  title="foo"
  >Text</div>

<div
  >Text</div>

<div
  >
  Text</div>

<div>Text</div
  >

<div
  />
```

From the rule tests at https://github.com/jacob-8/eslint-plugin-svelte-stylistic/blob/main/src/rules/brackets-same-line.test.ts

### Additional comments

I've already published it to npm as [eslint-plugin-svelte-stylistic](https://npmjs.com/package/eslint-plugin-svelte-stylistic) so I'm set, but I thought others might appreciate it so wanted to propose adding it here. What do you think? (rule 1 of 2, see second at #591)

### PR DESCRIPTION
**Pull request by mikededo**
Adds the `svelte/html-closing-bracket-newline` rule, which enforces that HTML
tags must have a newline (or not) after the closing bracket. This rule is
inspired by `vue/html-closing-bracket-newline`, and is implemented ensuring
what's discussed in #590.

Closes #590.

### CODE REVIEW

File: docs/rules/html-closing-bracket-new-line.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+/>
+<SelfClosing
+  multiline/>
+```
+
+</ESLintCodeBlock>
+
+## :wrench: Options
+
+```jsonc
```
Comment (mikededo) [inline docs/rules/html-closing-bracket-new-line.md:60]: @ota-meshi, `jsonc` feels like the right lang to use, does it have any issue with the docs syntax highlighting?

**Review by ota-meshi — APPROVED** [review]
Thank you for this PR! LGTM!

File: docs/rules/html-closing-bracket-new-line.md (hunk 1 above)
Comment (ota-meshi) [reply docs/rules/html-closing-bracket-new-line.md:63]: I think the syntax highlighting works fine on our website 👍

File: docs/rules/html-closing-bracket-new-line.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+<!-- prettier-ignore-end -->
+
+</ESLintCodeBlock>
+
+## :wrench: Options
+
+```jsonc
+{
+  "svelte/brackets-same-line": [
```
Comment (noy4) [inline docs/rules/html-closing-bracket-new-line.md:65]: should be `svelte/html-closing-bracket-new-line`?

File: docs/rules/html-closing-bracket-new-line.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+This rule enforces a line break (or no line break) before tag's closing brackets, which can also be configured to be enforced on self-closing tags.
+
+<ESLintCodeBlock fix>
+
+<!-- prettier-ignore-start -->
+<!--eslint-skip-->
+
+```svelte
+<script>
+  /* eslint svelte/brackets-same-line: "error" */
```
Comment (noy4) [inline docs/rules/html-closing-bracket-new-line.md:26]: `svelte/html-closing-bracket-new-line`?

**Comment by mikededo** [comment]
Hey @noy4, I've fixed the docs in #915. Thanks for pointing it out!

**Pull request merged**